### PR TITLE
Add PowerPC64 support for FreeBSD

### PR DESCRIPTION
**Pull request by lenoil98**
This initial set of changes are to add powerpc64 support for FreeBSD.  Additional work needs to be done to fix test build and failures.

### CODE REVIEW

**Comment by thewilsonator** [comment]
You have tabs everywhere, that cause build kite to fail.

**Comment by thewilsonator** [comment]
cc @redstar

**Comment by lenoil98** [comment]
I'll remove tabs and make adjustments.

**Comment by thewilsonator** [comment]
Thanks.

**Comment by thewilsonator** [comment]
If there isn't already (I'm pretty sure there is, there definitely is for X86) please add

```
version (PPC)
    version = PPC_Any;
else version (PPC64)
    version = PPC_Any;
```

and then use `version (PPC_Any)` to deduplicate the definitions. If you could also please clean up the commit history that would be great too.

File: src/core/sys/posix/ucontext.d [hunk 1]
```diff
@@ -1032,6 +1032,46 @@ else version (FreeBSD)
             ulong[8]        mc_spare;
         }
     }
+    else version (PPC)
+    {
+        alias uint   __register_t;
```
Comment (thewilsonator) [inline src/core/sys/posix/ucontext.d]: This is the only thing that is different between PPC and PPC64, please alias it to `size_t` and merge the two into a `version (PPC_Any)`.

**Pull request merged**